caiman/source_extraction/volpy/utils.py

```diff
@@ -89,61 +93,84 @@ def tellme(s):
 
     return ROIs
 
-def mrcnn_inference(img, size_range, weights_path, display_result=True):
-    """ Mask R-CNN inference in VolPy
-    Args: 
-        img: 2-D array
-            summary images for detection
-            
-        size_range: list
-            range of neuron size for selection
-            
-        weights_path: str
-            path for Mask R-CNN weight
+def mrcnn_inference_pytorch(img, size_range, weights_path, display_result=True):
+    """ 
+    Mask R-CNN inference in VolPy using PyTorch.
+    Args:
+        img (np.ndarray):
+            2-D or 3-D summary image for detection. If 2D, it's converted to 3-channel.
+
+        size_range (list):
+            Range of neuron size [min, max] for selection.
+
+        weights_path (str):
+            Path for the PyTorch Mask R-CNN weights file (.pt).
+
+        display_result (bool):
+            If True, the function will plot the result of the inference.
             
-        display_result: boolean
-            if True, the function will plot the result of inference
-        
-    Return:
-        ROIs: 3-D array
-            region of interests 
-            (# of components * # of pixels in x dim * # of pixels in y dim)
+        confidence_threshold (float):
+            The confidence threshold for accepting detected instances.
+
+    Returns:
+        ROIs: 3-D np.ndarray:
+            A 3-D array of ROIs (# of components, height, width).
     """
-    from caiman.source_extraction.volpy.mrcnn import visualize, neurons
-    import caiman.source_extraction.volpy.mrcnn.model as modellib
-    config = neurons.NeuronsConfig()
-    class InferenceConfig(config.__class__):
-        # Run detection on one img at a time
+    # Setup Configuration and Device
+    class InferenceConfig(Config):
```

Comment:
When testing this manually, I get a

`NameError: name 'Config' is not defined`